Display all the user reservations

AndreaM2429 opened the pull request:
Hello @HFG43! 🙋🏽‍♀️ 
I'm sending this PR to you because I implemented a new feature that requires your review.

# **🎯 Milestones:** 

- Create a `MyReservations` component to contain all the user reservations.
- Create a `ReservationCard` component that displays the information for each reservation, if the user doesn't have reservations the page displays a message.
- Each reservation has a `Cancel reservation` button that allows you to cancel it.
- Add styles.

## 🏁 General requirements 🏁:

❇️ Gitflow
❇️ No linter errors
❇️ Follow best practices

## ⚠️ Considerations

To test the functionality of the app please add the following import and routes in the `App component`. I didn't add them because they correspond to another task.

#
```ruby
import { Routes, Route } from 'react-router-dom';
import LoginPage from './Components/LoginPage';
import MainPage from './Components/MainPage';
import Register from './Components/Register';
import ReservationForm from './Components/ReservationForm';
import MyReservations from './Components/MyReservations';
``` 

Also, add the following code inside the `div` on the `line 19`

```ruby
<Routes>
    <Route path="/" element={<LoginPage />} />
    <Route path="/MainPage" element={<MainPage />} />
    <Route path="/Register" element={<Register />} />
    <Route path="/:experienceName/:experienceID" element={<ReservationForm />} />
    <Route path="/:userID/myReservations" element={<MyReservations />} />
 </Routes>
```

Please, add a `NavLink` in the `MainPage component` that redirects you to the `ReservationForm` and another for the `My reservations`.
You need at least one experience created with the `id = 1`.
You must add this because the `DetailsExperience` has not been created yet (it corresponds to another task).

```ruby
import { NavLink } from 'react-router-dom';

<NavLink to={`/${storageData.user.id}/myReservations`}>My reservations</NavLink>
<NavLink to="/Experince1/1">Book</NavLink>
```

☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆.｡.:･ﾟ☆

🙇‍♀️🙇‍♀️Thanks in advance Hernán for taking the time and all your effort to review these new implementations. If any change is required or you can see any improvements in the code please let me know on this PR.

AndreaM2429 committed:
Fetch the user reservations

AndreaM2429 committed:
Add function to give format a date

AndreaM2429 committed:
Add formats for the current reservations and cancel a reservation

AndreaM2429 committed:
Create the MyReservations component to display all the current reservations

AndreaM2429 committed:
Create the component to display the reservation info

AndreaM2429 committed:
Add styles

HFG43 reviewed (changes requested):
Hi Andre! @AndreaM2429 Great job so far! The app is starting to look fine! Please find below two print screens to support my comments.
![Screenshot from 2023-11-06 13-14-00](https://github.com/HFG43/final_capstone_front_end/assets/117187653/8770aa3f-17cb-4768-ab99-60ea1caa1dfa)

In Mobile design, I think is great and all reservations have the same size and proportion. I will consider for the `date` using just `yyyy-mm-dd`, as because we are not using the time for reservations, it may look unusual to have hour as `14:17`, just to give an example. Also, and it may be an issue from my side, but to review that it is uploading correctly the pictures from the URLs.

In the case of desktop version: 
![Screenshot from 2023-11-06 13-12-43](https://github.com/HFG43/final_capstone_front_end/assets/117187653/de170a4a-ffbf-4c08-aea0-52aa0b6e5307)

I think it will improve the UX to have the cards with the same size, and maybe considering placing more than 1 per line for a quick search

AndreaM2429 committed:
Fix styles

AndreaM2429 committed:
Fix date format

AndreaM2429 commented:
Hi @HFG43, thanks for your feedback and comments
I fixed the design and also changed the date format according to the considerations you provided me, Regarding the images there is a problem with some links because they render correctly, there is a screenshot to prove this.
![image](https://github.com/HFG43/final_capstone_front_end/assets/119966922/34bf1cd3-d997-420d-85fe-c36a5c455a1d)

HFG43 reviewed (approved):
@AndreaM2429 Great Andre! You did an excellent job implementing the changes. The project looks great!
There is nothing else to say than it's time to merge it!!
